crates/orderbook/src/lib.rs

```diff
@@ -41,7 +41,12 @@ pub fn serve_api(
     )
     .boxed();
     tracing::info!(%address, "serving order book");
-    let (_, server) = warp::serve(filter).bind_with_graceful_shutdown(address, shutdown_receiver);
+    let warp_svc = warp::service(filter);
+    let warp_svc = shared::make_service_with_task_local_storage!(warp_svc);
+    let server = hyper::Server::bind(&address)
+        .serve(warp_svc)
+        .with_graceful_shutdown(shutdown_receiver)
+        .map(|_| ());
```

Comment:
What does this map do?

Reply:
This map discards whatever result comes out of the server future and turns it into `()`.
Currently `serve_api` is expected to return `()` but we could also return an error here and adjust this [code](https://github.com/cowprotocol/services/blob/main/crates/orderbook/src/run.rs#L515-L518) to print that error.